src/golang/lib/notification/slack.go

```diff
@@ -26,13 +29,86 @@ func (s *SlackNotification) Level() shared.NotificationLevel {
 	return s.conf.Level
 }
 
+// reference: https://stackoverflow.com/questions/50106263/slack-api-to-find-existing-channel
+// We have to use list channel API together with a linear search.
+func findChannels(client *slack.Client, names []string) ([]slack.Channel, error) {
+	channels, _ /* cursor */, err := client.GetConversations(&slack.GetConversationsParameters{
+		ExcludeArchived: true,
+		Limit:           maxChannelLimit,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	namesSet := make(map[string]bool, len(names))
+	namesTaken := make(map[string]bool, len(names))
+	for _, name := range names {
+		namesSet[name] = true
+	}
+
+	// Slack channel names should be unique, based on
+	// https://twitter.com/slackhq/status/965358920146608128?lang=en .
+	// However, we make this list dynamic size on purpose to prevent potential
+	// compatibility issue in future. We will still send notifications to all
+	// channels matching the given name.
+	results := []slack.Channel{}
```

Comment:
I think we should be using `make(slack.Channel, 0, len(channels))` here instead. It's the same thing, but more canonical and makes the append more efficient.

Reply:
I guess I'm not entirely sure what you mean by "making the list of dynamic size on purpose".

Reply:
I thought `make()` will put a hard-limit on the capacity but it appears the size still extend if `append(` go beyonds the original capacity. So it should actually work!

Reply:
yeah it's the same thing but generally better since you can specify the capacity for it to pre-allocate.